synapse/storage/data_stores/main/e2e_room_keys.py

```diff
@@ -288,21 +370,31 @@ def _create_e2e_room_keys_version_txn(txn):
         )
 
     @trace
-    def update_e2e_room_keys_version(self, user_id, version, info):
+    def update_e2e_room_keys_version(
+        self, user_id, version, info=None, version_etag=None
+    ):
         """Update a given backup version
 
         Args:
             user_id(str): the user whose backup version we're updating
             version(str): the version ID of the backup version we're updating
             info(dict): the new backup version info to store
+            version_etag(int): tag of the keys in the backup
         """
+        updatevalues = {}
 
-        return self._simple_update(
-            table="e2e_room_keys_versions",
-            keyvalues={"user_id": user_id, "version": version},
-            updatevalues={"auth_data": json.dumps(info["auth_data"])},
-            desc="update_e2e_room_keys_version",
-        )
+        if info and "auth_data" in info:
+            updatevalues["auth_data"] = json.dumps(info["auth_data"])
+        if version_etag:
```

Comment:
surely it's not correct to handle zero differently to other `int`s?

```suggestion
        if version_etag is not None:
```